.github/workflows/reusable_deploy_v2_layer_stack.yml

```diff
@@ -126,8 +137,17 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # v3.5.3
+        with:
+          ref: ${{ env.RELEASE_COMMIT }}
+
+      - name: Restore sealed source code
+        uses: ./.github/actions/seal-restore
+        with:
+          integrity_hash: ${{ inputs.source_code_integrity_hash }}
+          artifact_name: ${{ inputs.source_code_artifact_name }}
+
       - name: Install poetry
-        run: pipx install poetry
+        run: pipx install git+https://github.com/python-poetry/poetry@68b88e5390720a3dd84f02940ec5200bfce39ac6 # v1.5.0
```

Comment:
I assume installing a non-versioned version of poetry causes issues?

Reply:
it opens up a supply chain attack where our build tool brings unexpected or malicious changes (+non-reproducible)